gtsam/base/types.h

```diff
@@ -230,3 +230,30 @@ namespace std {
 #ifdef ERROR
 #undef ERROR
 #endif
+
+namespace gtsam {
+
+  /**
+   * A trait to mark classes that need special alignment.
+   * This is required to make boost::make_shared and etc respect alignment, which is essential for the Python
+   * wrappers to work properly.
+   */
+#if __cplusplus < 201703L
+  template<typename ...> using void_t = void;
+#endif
+
+  template<typename, typename = void_t<>>
+  struct has_custom_allocator : std::false_type {
```

Comment:
The name `has_custom_allocator` is misleading. Really, above you are using `Eigen::aligned_allocator`.  I propose 

has_custom_allocator -> needs_eigen_aligned_allocator

_custom_allocator_type_trait -> _eigen_aligned_allocator_trait

Reply:
done